web/src/app/admin/languages/add-language.component.ts

```diff
@@ -17,78 +18,69 @@ export class AddLanguageComponent implements OnInit {
   sub!: Subscription;
   languages: ILanguage[] = [];
   isThere: Boolean = false;
-  languageToEdit: string;
+  languageToEdit: number;
   pageTitle: string;
-  language: ILanguage = {code: "code", name: "name", nativeName: "nativeName"};
+  
+  language: ILanguage = {id: 0, code: "code", name: "name", nativeName: "nativeName"};
 
-  constructor(private router: Router, private languageService: LanguageService, private data: EditLangugaeService) { }
+  constructor(private router: Router, private languageService: LanguageService, private data: EditLangugaeService, private snackBarService: SnackBarService) { }
 
-  clickAdd(name: string, nativeName: string, code: string, type: string): void {
-    //prototype method
-    if(type == "default") {
-      this.isThere = false;
-      for(let language of this.languages) {
-        if(language.name == name || language.nativeName == nativeName || language.code == code) {
-          this.isThere = true; break;
-        }
+  clickAdd(id: number, name: string, nativeName: string, code: string, type: number): void {
+    this.language = {id: (id+1), code: code, name: name, nativeName: nativeName}
+    //add
+    this.isThere = false;
+    for(let language of this.languages) {
+      if(language.name == name || language.nativeName == nativeName || language.code == code) {
+        this.isThere = true; break;
       }
-      if(!this.isThere) {
-        if(name!="" && nativeName!="" && code!="") {
-          this.languages.push({name, nativeName, code});
-          alert('New Language added');
+    }
+    if(!this.isThere) {
+      if(type == 0) {
+        if(this.language.name!="" && this.language.nativeName!="" && this.language.code!="") {
+          this.languageService.createLanguage(this.language)
+          .subscribe({
+          next: () => this.onSaveComplete(),
+          error: err => { this.errorMessage = err; console.log(this.errorMessage); }
+          });
         } else {
-          alert('Please fill every field')
+          this.snackBarService.warn('Please fill every field');
         }
+    //edit
       } else {
-        alert('Some Properties of your language already exist!');
-      }
-  
-      //test output -- can be removed later
-      for(let language of this.languages) {
-        console.log(language.name);
-      }
-      //get notification from server
-      //if saved show confirmation message
-      //if existing show error
-    } else {
-      for(let language of this.languages) {
-        if(language.name == this.languageToEdit) {
-          if(name!="" && nativeName!="" && code!="") {
-            language.name = name;
-            language.nativeName = nativeName;
-            language.code = code;
-            alert('Language changed');
-          } else {
-            alert('Please fill every field')
+        for(let language of this.languages) {
```

Comment:
I think you might don't need to validate the id nor the language handle its generation. The generation of the id will be done in the backend, the mock app auto generate this as well. 
```suggestion
        for(let language of this.languages) {
```